From 65b9b64c8a243e33aa6d22624a0d3e7818d062f1 Mon Sep 17 00:00:00 2001
From: Xi Ruoyao <xry111@xry111.site>
Date: Wed, 10 Jun 2026 21:45:16 +0800
Subject: [PATCH 16/18] refactor: make zx_fbdev_set_suspend no-op on Linux >=
 6.15

On Linux >= 6.15 we use FB console based on DRM client, which already
handles the suspend/resume in a generic way.  So we don't need to
duplicate the logic.

Signed-off-by: Xi Ruoyao <xry111@xry111.site>
---
 src/zx_fbdev.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/zx_fbdev.c b/src/zx_fbdev.c
index a78454a..ef01a91 100644
--- a/src/zx_fbdev.c
+++ b/src/zx_fbdev.c
@@ -432,6 +432,7 @@ int zx_fbdev_deinit(zx_card_t *zx)
 
 void zx_fbdev_set_suspend(zx_card_t *zx, int state)
 {
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 15, 0)
     struct zx_fbdev *fbdev = zx->fbdev;
     struct fb_info *info;
 
@@ -456,6 +457,7 @@ void zx_fbdev_set_suspend(zx_card_t *zx, int state)
 
         console_unlock();
     }
+#endif
 }
 
 void zx_fbdev_poll_changed(struct drm_device *dev)
-- 
2.54.0

